service/frontend/handler.go

```diff
@@ -360,11 +384,35 @@ func (wh *WorkflowHandler) PollForDecisionTask(
 
 	wh.Service.GetLogger().Debugf("Poll for decision. DomainName: %v, DomainID: %v", domainName, info.ID)
 
+	pollerID := uuid.New()
 	matchingResp, err := wh.matching.PollForDecisionTask(ctx, &m.PollForDecisionTaskRequest{
 		DomainUUID:  common.StringPtr(info.ID),
+		PollerID:    common.StringPtr(pollerID),
 		PollRequest: pollRequest,
 	})
 	if err != nil {
+		// First check if this err is due to context cancellation.  This means client connection to frontend is closed.
+		if ctx.Err() == context.Canceled {
+			// Our rpc stack does not propagates context cancellation to the other service.  Lets make an explicit
```

Comment:
can you move this cancel logic into its own method ?

Reply:
Done.